src/handlers/delete_client.rs

```diff
@@ -1,49 +1,46 @@
 use crate::handlers::{new_error_response, new_success_response, ErrorReason};
-use crate::State;
+use crate::AppState;
+use axum::extract::{Path, State};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::Json;
+use serde_json::json;
 use std::sync::Arc;
-use warp::http;
 
 pub async fn handler(
-    id: String,
-    state: Arc<State<impl crate::store::ClientStore>>,
-) -> Result<impl warp::Reply, warp::Rejection> {
+    Path(id): Path<String>,
+    State(state): State<Arc<AppState<impl crate::store::ClientStore>>>,
+) -> impl IntoResponse {
```

Comment:
I'd suggest refactoring the return type to something like `Result<(), HandlerError>` and then implement `impl<T: Serialize> IntoResponse for Result<T, HandlerError>` which would serialize the data into JSON and specify the status code. Doing this would allow you to handle errors naturally using `?` inside your handler functions, and clean up the JSON serialization.